core/primitives/src/shard_layout.rs

```diff
@@ -1353,4 +1389,94 @@ mod tests {
             )
         );
     }
+
+    #[test]
+    fn test_deriving_shard_layout() {
+        macro_rules! boundaries {
+            ($($b:expr),*) => {
+                vec![$($b.parse().unwrap()),*]
+            };
+        }
+
+        macro_rules! shards {
+            ($($s:expr),*) => {
+                vec![$(ShardId::new($s as u64)),*]
+            };
+        }
```

Comment:
This is a bit too crazy for me - can you rewrite it in simpler rust please?

Reply:
Yeah, these may be overkill..